src/snowflake/snowpark/stored_procedure.py

```diff
@@ -102,8 +103,13 @@ def __call__(
 
         if self._anonymous_sp_sql:
             call_sql = generate_call_python_sp_sql(session, self.name, *args)
-            df = session.sql(f"{self._anonymous_sp_sql}{call_sql}")
+            query = f"{self._anonymous_sp_sql}{call_sql}"
+            df = session.sql(query)
             if self._is_return_table:
+                qid = session._conn.execute_and_get_sfqid(
+                    query, statement_params=statement_params
+                )
+                df = session.sql(result_scan_statement(qid))
```

Comment:
The return-type of this function is incorrect. It should be `Any` and not `any`. The lower-case version is a builtin. I know this isn't your doing, but could you please fix it?